sympy/integrals/risch.py

```diff
@@ -162,7 +173,8 @@ class DifferentialExtension(object):
     # only create one DifferentialExtension per integration).  Also, it's nice
     # to have a safeguard when debugging.
     __slots__ = ('f', 'x', 'T', 'D', 'fa', 'fd', 'Tfuncs', 'backsubs', 'E_K',
-        'E_args', 'L_K', 'L_args', 'cases', 'case', 't', 'd', 'newf', 'level',
+        'E_args', 'L_K', 'L_args', 'T_K', 'T_args', 'AT_K', 'AT_args', 'ALG_K',
```

Comment:
> two nested lists

The lists should not be nested in the technical sense but rather 'shuffled' in the same way as the `cases` list that contains the type of `i`-th extensions at slot `i`. So, the first list would contain 'log' or 'atan' where `cases` has 'primitive'. (Until now, the only 'primitive' has been 'log'.)